hazelcast-jet-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/map/IMapSqlConnector.java

```diff
@@ -122,6 +132,69 @@ public final Table createTable(
         );
     }
 
+    @Override
+    public boolean supportsNestedLoopReader() {
+        return true;
+    }
+
+    @Nonnull @Override
+    public Vertex nestedLoopReader(
+            @Nonnull DAG dag,
+            @Nonnull Table table0,
+            @Nullable Expression<Boolean> predicate,
+            @Nonnull List<Expression<?>> projections,
+            @Nonnull JetJoinInfo joinInfo
+    ) {
+        PartitionedMapTable table = (PartitionedMapTable) table0;
+
+        ProcessorSupplier pSupplier;
+        String name = table.getMapName();
+        List<TableField> fields = table.getFields();
+        QueryPath[] paths = fields.stream().map(field -> ((MapTableField) field).getPath()).toArray(QueryPath[]::new);
+        QueryDataType[] types = fields.stream().map(TableField::getType).toArray(QueryDataType[]::new);
+        QueryTargetDescriptor keyDescriptor = table.getKeyDescriptor();
+        QueryTargetDescriptor valueDescriptor = table.getValueDescriptor();
+
+        KvRowProjector.Supplier rightRowProjectorSupplier =
+                KvRowProjector.supplier(paths, types, keyDescriptor, valueDescriptor, predicate, projections);
+
+        String desc;
+        if (isEquiJoinByPrimitiveKey(joinInfo, fields)) {
+            pSupplier = new JoinByPrimitiveKeyProcessorSupplier(joinInfo, name, rightRowProjectorSupplier);
```

Comment:
We should somehow use a distributed-partitioned edge in this case. The `map.get` call can then be local.